test/e2e/network/service.go

```diff
@@ -2697,11 +2698,21 @@ func createPodOrFail(f *framework.Framework, ns, name string, labels map[string]
 }
 
 // launchHostExecPod launches a hostexec pod in the given namespace and waits
-// until it's Running
+// until it's Running. It retries creation if the pod exists with the given name
+// and is being deleted.
 func launchHostExecPod(client clientset.Interface, ns, name string) *v1.Pod {
 	framework.Logf("Creating new host exec pod")
 	hostExecPod := e2epod.NewExecPodSpec(ns, name, true)
-	pod, err := client.CoreV1().Pods(ns).Create(context.TODO(), hostExecPod, metav1.CreateOptions{})
+	var pod *v1.Pod
+	err := wait.PollImmediate(framework.Poll, framework.PodStartTimeout, func() (bool, error) {
+		var err error
+		pod, err = client.CoreV1().Pods(ns).Create(context.TODO(), hostExecPod, metav1.CreateOptions{})
+		// TODO: we should have a specific cause for "AlreadyExists"
+		if err != nil && apierrors.IsAlreadyExists(err) && strings.Contains(err.Error(), "is being deleted") {
+			return false, nil
+		}
+		return true, err
```

Comment:
we use to retry on all errors, ie. "I can't connect to apiserver" or "apiserver returns 500 because etcd ... " or "connection timeout"
Maybe just logging a message only for those specific errors you are looking for ?

Reply:
Not sure what you mean?  This retries *only* alreadyexists, and reports the same error (return true, err ensures that if an error happens it goes to the next line)

Reply:
maybe I'm not understanding it well, but I think that if there is an error, that is not "alreadyExists", this will fail the test.

Reply:
ahhhh ok, nice, I got it, we were trying to do this in another test.
Scratch my comment